src/weapon/w_001.c

```diff
@@ -13,7 +13,162 @@ extern s32 g_HandId;
 
 INCLUDE_ASM("weapon/nonmatchings/w_001", EntityWeaponAttack);
 
-INCLUDE_ASM("weapon/nonmatchings/w_001", func_ptr_80170004);
+extern WeaponAnimation D_B000_8017AF44[];
+extern uvPair D_B000_8017AFB8[];
+
+s32 func_ptr_80170004(Entity* self) {
+    Primitive* prim;
+    s32 i;
+    u8 handOffset;
+    s16 x, y;
+    s16 xOffset, yOffset;
+    s16 posX, posY;
+    uvPair* p;
+    s16 animIndex;
+    WeaponAnimation* anim;
+
+    handOffset = 0;
+    if (g_HandId) {
+        handOffset = 1 << 7;
```

Comment:
I think this is better as 0x80, since it's being used as a number (moving the V coordinate) rather than as a bit flag (moving a 1 seven positions to the left).